File: planes/phase-dp2-nats/create-streams.py
#!/usr/bin/env python3
import asyncio
import nats
import json

async def main():
    # Connect to NATS
    nc = await nats.connect("nats://nats:4222")
    
    # Create JetStream context
    js = nc.jetstream()
    
    # Stream configurations
    streams = [
        {
            "name": "DOCUMENTS",
            "subjects": ["data.doc.>"],
            "retention": "workqueue",
            "max_msgs": 100000,
            "max_bytes": 5368709120,  # 5GB
            "storage": "file",
            "num_replicas": 1,
            "discard": "old",
            "duplicate_window": 120,  # seconds
            "max_msg_size": 1048576,  # 1MB
            "max_age": 0,
            "max_msgs_per_subject": -1,
            "allow_rollup": False,
            "deny_delete": False,
            "deny_purge": False,
            "description": "Document processing stream with work queue retention"
        },
        {
            "name": "EXECUTION",
            "subjects": ["exec.task.>"],
            "retention": "interest",
            "max_age": 86400,  # 24 hours in seconds
            "storage": "file",
            "num_replicas": 1,
            "discard": "old",
            "duplicate_window": 60,  # seconds
            "max_msg_size": 524288,  # 512KB
            "max_bytes": 2147483648,  # 2GB
            "max_msgs": 50000,
            "max_msgs_per_subject": -1,
            "allow_rollup": False,
            "deny_delete": False,
            "deny_purge": False,
            "description": "Task execution stream with 24h retention, 2GB limit"
        },
        {
            "name": "OBSERVABILITY",
            "subjects": ["obs.metric.>"],
            "retention": "limits",
            "max_bytes": 1073741824,  # 1GB
            "storage": "file",
            "num_replicas": 1,
            "discard": "old",
            "duplicate_window": 30,  # seconds
            "max_msg_size": 131072,  # 128KB
            "max_age": 0,
            "max_msgs_per_subject": -1,
            "allow_rollup": False,
            "deny_delete": False,
            "deny_purge": False,
            "description": "Observability metrics stream with size limits"
        }
    ]
    
    # Create each stream
    for stream_config in streams:
        try:
            print(f"Creating stream: {stream_config['name']}")
            await js.add_stream(**stream_config)
            print(f"  ✓ Created {stream_config['name']}")
        except Exception as e:
            print(f"  ✗ Error creating {stream_config['name']}: {e}")
    
    # List streams to verify
    print("\nVerifying streams...")
    streams_info = await js.streams_info()
    for stream in streams_info:
        print(f"  - {stream.config.name}: {stream.state.messages} messages, {stream.state.bytes} bytes")
    
    await nc.close()

if __name__ == "__main__":
    asyncio.run(main())

File: planes/phase-dp2-nats/create-streams.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"github.com/nats-io/nats.go"
)

func main() {
	// Connect to NATS
	nc, err := nats.Connect("nats://nats:4222")
	if err != nil {
		log.Fatalf("Failed to connect: %v", err)
	}
	defer nc.Close()

	// Create JetStream context
	js, err := nc.JetStream()
	if err != nil {
		log.Fatalf("Failed to create JetStream context: %v", err)
	}

	// Stream configurations
	streams := []nats.StreamConfig{
		{
			Name:        "DOCUMENTS",
			Subjects:    []string{"data.doc.>"},
			Retention:   nats.WorkQueuePolicy,
			MaxMsgs:     100000,
			MaxBytes:    5 * 1024 * 1024 * 1024, // 5GB
			Storage:     nats.FileStorage,
			NumReplicas: 1,
			Discard:     nats.DiscardOldPolicy,
			Description: "Document processing stream with work queue retention",
		},
		{
			Name:        "EXECUTION",
			Subjects:    []string{"exec.task.>"},
			Retention:   nats.InterestPolicy,
			MaxMsgs:     50000,
			MaxBytes:    2 * 1024 * 1024 * 1024, // 2GB
			MaxAge:      24 * time.Hour,
			Storage:     nats.FileStorage,
			NumReplicas: 1,
			Discard:     nats.DiscardOldPolicy,
			Description: "Task execution stream with 24h retention, 2GB limit",
		},
		{
			Name:        "OBSERVABILITY",
			Subjects:    []string{"obs.metric.>"},
			Retention:   nats.LimitsPolicy,
			MaxBytes:    1 * 1024 * 1024 * 1024, // 1GB
			Storage:     nats.FileStorage,
			NumReplicas: 1,
			Discard:     nats.DiscardOldPolicy,
			Description: "Observability metrics stream with size limits",
		},
	}

	// Create each stream
	for _, stream := range streams {
		fmt.Printf("Creating stream: %s...\n", stream.Name)
		
		// Check if stream already exists
		info, err := js.StreamInfo(stream.Name)
		if err == nil && info != nil {
			fmt.Printf("  Stream %s already exists\n", stream.Name)
			continue
		}
		
		_, err = js.AddStream(&stream)
		if err != nil {
			fmt.Printf("  Error creating stream %s: %v\n", stream.Name, err)
			continue
		}
		fmt.Printf("  ✓ Created stream %s\n", stream.Name)
	}

	// Verify streams
	fmt.Println("\nVerifying streams...")
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()

	streamsInfo, err := js.StreamsInfo()
	if err != nil {
		log.Fatalf("Failed to get streams info: %v", err)
	}

	for _, s := range streamsInfo {
		fmt.Printf("  - %s: %d messages, %d bytes\n", s.Config.Name, s.State.Messages, s.State.Bytes)
	}

	fmt.Println("\nDone!")
}